platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java

```diff
@@ -48,81 +42,44 @@
 
-    private static final String CONFIG_NAME = "ldap";
-
     private static final String SUFFICIENT_FLAG = "sufficient";
 
     private static final Logger LOGGER = LoggerFactory.getLogger(LdapLoginConfig.class);
 
     private static final String LDAP_MODULE = ddf.ldap.ldaplogin.SslLdapLoginModule.class.getName();
 
-    private ServiceRegistration<JaasRealm> registration = null;
+    private String id = UUID.randomUUID()
+            .toString();
 
     private Map<String, Object> ldapProperties = new HashMap<>();
 
-    /**
-     * Registers the passed-in modules under a new JaasRealm.
-     *
-     * @param modules
-     *            Modules to add to the JaasRealm.
-     */
-    private void registerConfig(Module[] modules) {
-        if (registration != null) {
-            LOGGER.debug("Found previous registration, unregistering old service.");
-            try {
-                registration.unregister();
-            } catch (IllegalStateException ise) {
-                LOGGER.debug(
-                        "Service was already unregistered, continuing on to register new service.");
-            }
-        }
-        Config config = new Config();
-        BundleContext context = getContext();
-        config.setBundleContext(context);
-        config.setName(CONFIG_NAME);
-        config.setRank(2);
-        config.setModules(modules);
-        LOGGER.debug("Registering new service as a JaasRealm.");
-        if (context != null) {
-            registration = context.registerService(JaasRealm.class, config, null);
-        }
-    }
-
-    protected BundleContext getContext() {
-        Bundle cxfBundle = FrameworkUtil.getBundle(LdapLoginConfig.class);
-        if (cxfBundle != null) {
-            return cxfBundle.getBundleContext();
-        }
-        return null;
-    }
+    private LdapService ldapService;
 
     /**
      * Update method that receives new properties.
      *
-     * @param props
-     *            Map of properties.
+     * @param props Map of properties.
      */
     public void update(Map<String, ?> props) {
         if (props != null) {
             LOGGER.debug("Received an updated set of configurations for the LDAP Login Config.");
             // create modules from the newly updated config
             Module ldapModule = createLdapModule(props);
-            registerConfig(new Module[] {ldapModule});
+            ldapService.update(ldapModule);
         }
     }
 
     /**
      * Creates a new module with the given properties.
      *
-     * @param properties
-     *            Map of properties.
+     * @param properties Map of properties.
      * @return newly created module.
      */
     private Module createLdapModule(Map<String, ?> properties) {
         Module ldapModule = new Module();
         ldapModule.setClassName(LDAP_MODULE);
         ldapModule.setFlags(SUFFICIENT_FLAG);
-        ldapModule.setName("ldapModule");
+        ldapModule.setName(id);
```

Comment:
Is there any advantage from a human-readability/administration perspective in prefixing this UUID with a constant string like 'LDAP:'?